src/test/java/com/avispl/symphony/dal/communicator/other/genericserver/WebClientCommunicatorTest.java

```diff
@@ -0,0 +1,204 @@
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 200 OK for Jpg file
+	 */
+	@Test
+	public void getMultipleStatisticsWithJpgFile() throws Exception {
+		// Attempt to check accessible for a jpg file
+		webClientCommunicator.setURI("/jpg-file.jpg");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("200 OK", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 403 Forbidden.
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode403() throws Exception {
+		//  Expect throw API Error
+		webClientCommunicator.setURI("/forbidden");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("403 Forbidden", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with API Error due to 404 response status code.
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode404() {
+		//  Expect throw API Error
+		webClientCommunicator.setURI("/not-exist-uri");
+		try {
+			ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+			fail("Expect fail here due to status code 404 out of range 200");
+		} catch (Exception e) {
+			// expected
+			assertEquals("GET operation failed", e.getMessage());
+		}
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with default case that does not configure URI.
+	 */
+	@Test
+	public void getMultipleStatisticsWithDefault() throws Exception {
+		// should be "Not Configured" when the URI are not set in Adapter Properties
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("Not Configured", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response 401 Unauthorized
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode401() throws Exception {
+		//  Should be "401 Unauthorized" when the URI are not set in Adapter Properties
+		webClientCommunicator.setURI("invalid-login");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("401 Unauthorized", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with API Error due to 500 response status code.
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode500() {

```

Comment:
Same here about the 
`@Test(expected = SomeError.class)` :)

Reply:
Since we're using junit jupiter, there is no expected in @Test as before, so move to use assertThrows instead.